packages/trpc/server/document-router/router.ts

```diff
@@ -172,4 +173,66 @@ export const documentRouter = router({
         });
       }
     }),
+
+  duplicateDocument: authenticatedProcedure
+    .input(ZGetDocumentByIdQuerySchema)
+    .mutation(async ({ input, ctx }) => {
+      try {
+        const { id } = input;
```

Comment:
Extract logic into `packages/lib/server-only/document/duplicate-document-by-id`

We don't want to keep any logic in our tRPC router so we can easily reuse it when implementing the REST and GraphQL API's 😄